app/app.go

```diff
@@ -705,7 +706,7 @@ func (app *OKExChainApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBloc
 
 // EndBlocker updates every end block
 func (app *OKExChainApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
-	if appconfig.GetOecConfig().GetDynamicGpMode() != types.CloseMode {
+	if appconfig.GetOecConfig().GetDynamicGpMode() != types.MinimalGpMode {
```

Comment:
It may be crash, when set init MinimalGpMode, but set dynamic not MinimalGpMode

Reply:
Fixed. When MinimalGpMode is set at initialization, the app also initializes the "app.gpo". And when DeliverTx or ParallelTxs, all modes will collect the gasPrice and gasUsed of tx.